### PR TITLE
Update the NoSQL image and documentation

### LINKED ISSUE #2365: Increasing NoSQL KVLite storage size
Hi There,

Thanks for the nice docker images. I'm testing out the NoSQL image and noticed this output `...available storage size:1023 MB...` when running this command as in the README:

```bash
docker run --rm -ti --link kvlite:store oracle/nosql:ce \
  java -jar lib/kvstore.jar runadmin -host store -port 5000 -store kvstore

kv-> ping
```

Does this mean there's only 1023 MB of storage for any tables created? Is there a way to increase this (e.g., setting some flag)?

Thanks for any help!

### PR DESCRIPTION
**Pull request by dario-vega**
This PR updates the Oracle NoSQL Dockerfile to align with the latest version and bundles.
- this image will use the latest version kv-ce-21.2.46.zip.

This PR updates with some improvements to the Oracle NoSQL CE image.
- fixes #2365 - The CE 21.2 version introduces a new parameter storagedirsizegb

Signed-off-by: Dario Vega dario.vega@oracle.com

### CODE REVIEW

**Review by mikebrey — APPROVED** [review]

File: NoSQL/README.md [hunk 1]
```diff
@@ -8,30 +8,20 @@ This project offers sample container image configuration files for:
 
 * [Oracle NoSQL Database Community Edition](ce/Dockerfile)
 
-## Quick start: building the Oracle NoSQL Community Edition image
+This container image uses KVLite. **KVLite is a simplified version of the Oracle NoSQL Database**. It provides a single storage node, a single shard store, with no
+replication and no administration. It runs in a single process. **Note:** KVLite is not intended for production deployment or performance measurements.
```
Comment (Djelibeybi) [inline NoSQL/README.md:12]:
```suggestion
This container image uses a simplified version of the Oracle NoSQL Database called KVLite. KVLite runs as a single process that provides a single storage node and single storage shard. KVLite does not include replication or administration. 

> **Note:** KVLite is not intended for production deployment or performance measurements.
```

File: NoSQL/README.md [hunk 2]
```diff
@@ -44,6 +34,8 @@ KVLite is the default `CMD` of the image:
 ```shell
 docker run -d --name=kvlite --hostname=kvlite --env KV_PROXY_PORT=8080 -p 8080:8080 oracle/nosql:ce
 ```
+**Note**:  By default, the KVLite store created has a size of `10GB`, use `--env KV_STORAGESIZE=N` to override that. 
+The value of N is in GB and must be >= 1
```
Comment (Djelibeybi) [inline NoSQL/README.md:38]:
```suggestion
The value of `N` is in gigabytes and must be greater than 1GB.
```

File: NoSQL/README.md [hunk 3]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+
+You will use you current HOSTNAME as a value for the --hostname
+
+```shell
+docker run -d --name=kvlite --hostname=$HOSTNAME --env KV_PROXY_PORT=8080 -p 8080:8080 \
+-p 5000:5000 -p 5010-5020:5010-5020 -p 5021-5049:5021-5049 -p 5999:5999 oracle/nosql:ce
+```
+
+**Note**:  By default, the KVLite store created has a size of `10GB`, use `--env KV_STORAGESIZE=N` to override that. 
+The value of N is in GB and must be >= 1
```
Comment (Djelibeybi) [inline NoSQL/README.md:175]:
```suggestion
The value of `N` is in gigabytes and must be greater than 1GB.
```

File: NoSQL/README.md [hunk 4]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+```
+Note: -host must be the same name used when starting the container
+
+If you want to run the NoSQL command to ping the kvlite store from another container:
+
+```shell
+docker run --rm -ti --link kvlite:store oracle/nosql:ce \
+  java -jar lib/kvstore.jar ping -host store -port 5000
+```
+Note the required use of --link for proper hostname check (actual KVLite container is named kvlite; alias is store).
```
Comment (Djelibeybi) [inline NoSQL/README.md:191]:
```suggestion
Note the required use of `--link` for proper hostname check (actual KVLite container is named `kvlite` and its alias is `store`).
```

File: NoSQL/README.md [hunk 5]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+**Note**:  By default, the KVLite store created has a size of `10GB`, use `--env KV_STORAGESIZE=N` to override that. 
+The value of N is in GB and must be >= 1
+
+In a second shell, run the NoSQL command to ping the kvlite store
+instance:
+
+```shell
+java -jar $KVHOME/lib/kvstore.jar ping -host $HOSTNAME -port 5000
+```
+Note: -host must be the same name used when starting the container
```
Comment (Djelibeybi) [inline NoSQL/README.md:183]:
```suggestion
Note: the value provided for `-host` must match the name used when starting the container.
```

File: NoSQL/README.md [hunk 6]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+Start up KVLite in a container. You must give it a name and provide a hostname. 
+In this case, You need to publish all internal ports and the KV_PROXY_PORT.
+- 5000 KVPORT
+- 5010-5020 KV_HARANGE
+- 5021-5049 KV_SERVICERANGE
+- 8080 KV_PROXY_PORT
+
+Startup of KVLite is the default `CMD` of the image:
+
+You will use you current HOSTNAME as a value for the --hostname
```
Comment (Djelibeybi) [inline NoSQL/README.md:167]:
```suggestion
Use the environment variable `$HOSTNAME` as the value for the `--hostname`
```

File: NoSQL/README.md [hunk 7]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+DOWNLOAD_LINK="${DOWNLOAD_ROOT}/${DOWNLOAD_FILE}"
+curl -OLs $DOWNLOAD_LINK
+jar tf $DOWNLOAD_FILE | grep "kv-$KV_VERSION/lib" > extract.libs
+jar xf $DOWNLOAD_FILE @extract.libs 
+rm -f $DOWNLOAD_FILE extract.libs
+KVHOME=$PWD/kv-$KV_VERSION
+```
+
+Start up KVLite in a container. You must give it a name and provide a hostname. 
+In this case, You need to publish all internal ports and the KV_PROXY_PORT.
```
Comment (Djelibeybi) [inline NoSQL/README.md:159]:
```suggestion
In this case, You need to publish all internal ports and the `KV_PROXY_PORT`:
```

File: NoSQL/README.md [hunk 8]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+- 5000 KVPORT
+- 5010-5020 KV_HARANGE
+- 5021-5049 KV_SERVICERANGE
+- 8080 KV_PROXY_PORT
```
Comment (Djelibeybi) [inline NoSQL/README.md:163]:
```suggestion

- 5000 `KVPORT`
- 5010-5020 `KV_HA_RANGE` 
- 5021-5049 `KV_SERVICE_RANGE`
- 8080 `KV_PROXY_PORT`
```

File: NoSQL/README.md [hunk 9]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
 });
 ````
 
-## Using Oracle NoSQL Command-Line from an external host
+## Advanced Scenario: Using Oracle NoSQL Command-Line from an external host
+
+**Note**: We recommend running NoSQL Command-Line doing a container to container connection as shown in the previous chapters. 
+It allows starting the container without publishing all internal ports (KVPORT, KV_HARANGE, KV_SERVICERANGE) but only the KV_PROXY_PORT.
```
Comment (Djelibeybi) [inline NoSQL/README.md:137]:
```suggestion
**Note**: We recommend running NoSQL Command-Line and creating a container-to-container connection as detailed above.. 

It allows you to start a container with only the `KV_PROXY_PORT` enabled. 
```

File: NoSQL/README.md [hunk 10]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+
+In a second shell, run the NoSQL command to ping the kvlite store
+instance:
+
+```shell
+java -jar $KVHOME/lib/kvstore.jar ping -host $HOSTNAME -port 5000
+```
+Note: -host must be the same name used when starting the container
+
+If you want to run the NoSQL command to ping the kvlite store from another container:
```
Comment (Djelibeybi) [inline NoSQL/README.md:185]:
```suggestion
If you want to run the NoSQL command to ping the KVLite store from another container:
```

File: NoSQL/README.md [hunk 11]
```diff
@@ -129,7 +131,80 @@ return new NoSQLClient({
...
+
+```shell
+docker run --rm -ti --link kvlite:store oracle/nosql:ce \
+  java -jar lib/kvstore.jar ping -host store -port 5000
+```
+Note the required use of --link for proper hostname check (actual KVLite container is named kvlite; alias is store).
+
+If you want to run without --link, you cannot use any alias when starting the container (use HOSTNAME).
```
Comment (Djelibeybi) [inline NoSQL/README.md:193]:
```suggestion
If you want to run without `--link`, you can not use any alias when starting the container, so use `$HOSTNAME` instead..  
```

File: NoSQL/README.md [hunk 12]
```diff
@@ -179,27 +254,15 @@ PING kvlite-nosql-container-host (10.0.0.143) 56(84) bytes of data.
...
-
 ```shell
 docker run -d --name=kvlite --hostname=kvlite-nosql-container-host --env KV_PROXY_PORT=8080 -p 8080:8080 \
 -p 5000:5000 -p 5010-5020:5010-5020 -p 5021-5049:5021-5049 -p 5999:5999 oracle/nosql:ce
 ```
-
-In a second shell, run the NoSQL command to ping the kvlite store
-instance:
+**Note**:  By default, the KVLite store created has a size of `10GB`, use `--env KV_STORAGESIZE=N` to override that. 
+The value of N is in GB and must be >= 1
```
Comment (Djelibeybi) [inline NoSQL/README.md:262]:
```suggestion
The value of `N` is in gigabytes and must be greater than 1GB.
```

File: NoSQL/README.md [hunk 13]
```diff
@@ -179,27 +254,15 @@ PING kvlite-nosql-container-host (10.0.0.143) 56(84) bytes of data.
...
 
 ```shell
-$ java -jar $KVHOME/lib/kvstore.jar ping -host kvlite-nosql-container-host -port 5000
+java -jar $KVHOME/lib/kvstore.jar ping -host kvlite-nosql-container-host -port 5000
 ```
 Note: -host must be the same name used when starting the container
```
Comment (Djelibeybi) [inline NoSQL/README.md:267]:
```suggestion
Note: `-host` must be the same name used when starting the container
```

File: NoSQL/README.md [hunk 14]
```diff
@@ -216,26 +279,46 @@ If you want to run without --link, you cannot use any alias when starting the co
...
+```
+
+The resulting image will be available as `oracle/nosql:ce`. 
+
+
 ## More information
 
 For more information on [Oracle NoSQL](http://www.oracle.com/technetwork/database/database-technologies/nosqldb/overview/index.html)
-please review the [product documentation](http://docs.oracle.com/cd/NOSQL/html/index.html).
+please review the [product documentation](https://docs.oracle.com/en/database/other-databases/nosql-database/index.html).
```
Comment (Djelibeybi) [inline NoSQL/README.md:314]:
```suggestion
please review the [Oracle NoSQL Database product documentation](https://docs.oracle.com/en/database/other-databases/nosql-database/index.html).
```

File: NoSQL/README.md [hunk 15]
```diff
@@ -216,26 +279,46 @@ If you want to run without --link, you cannot use any alias when starting the co
...
 
 The Oracle NoSQL Database Community Edition image contains the OpenJDK.
 
 ## Licenses
 
-Oracle NoSQL Community Edition is licensed under the [APACHE LICENSE v2.0](https://docs.oracle.com/cd/NOSQL/html/driver_table_c/doc/LICENSE.txt).
+Oracle NoSQL Community Edition is licensed under the [APACHE LICENSE v2.0](https://docs.oracle.com/en/database/other-databases/nosql-database/22.1/license/index.html#NSXLI-GUID-006E432E-1965-45A2-AEDE-204BD05E1560) .
```
Comment (Djelibeybi) [inline NoSQL/README.md:320]:
```suggestion
Oracle NoSQL Community Edition is released under the [Apache 2.0 License](https://docs.oracle.com/en/database/other-databases/nosql-database/22.1/license/index.html#NSXLI-GUID-006E432E-1965-45A2-AEDE-204BD05E1560) . 
```

**Review by Djelibeybi — CHANGES_REQUESTED** [review]

**Review by mikebrey — DISMISSED** [review]
changes all look reasonable

**Review by Djelibeybi — DISMISSED** [review]

**Review by Djelibeybi — APPROVED** [review]

**Review by mikebrey — APPROVED** [review]
approved

**Pull request merged**